File: javascript/module1/challenges/challenge3.js
function weaponDamage(weaponBaseDamage, playerLevelOfSkill) {
  // damage is the base weapon damage, plus the skill factor, plus random factor
  let damage =
    weaponBaseDamage +
    weaponBaseDamage * (playerLevelOfSkill * 0.05) +
    Math.random() * 10;

  // damage can't be less than 20
  return Math.max(damage, 20);
}

File: csharp/module1/challenges/challenge3.cs
public double WeaponDamage(double weaponBaseDamage, int playerLevelOfSkill) {
    Random random = new Random();
    double damage = weaponBaseDamage + weaponBaseDamage * (playerLevelOfSkill * 0.05) + random.NextDouble() * 10;
    return Math.Max(damage, 20);
}
